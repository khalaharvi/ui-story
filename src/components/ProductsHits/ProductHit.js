import React from 'react';
import PropTypes from 'prop-types';

import Image from 'components/Image';
import { cx } from './ProductsHits';

// TO DO: add fallback images
const getDisplayPrice = (price) => {
  let displayPrice = 'Price on Request';

  if (price) {
    displayPrice = `$${price.toLocaleString()}`;
  }

  return displayPrice;
};

const ProductHit = ({ hit, transformHitImageUrl }) => (
  <div>
    <div className={cx('list-item-img')}>
      <Image
        src={
          transformHitImageUrl ?
          transformHitImageUrl(hit.primary_image_url) :
          hit.primary_image_url
        }
        alt={hit.title}
      />
    </div>
    <b className={cx('list-item-title')}>{hit.title}</b>
    <i className={cx('list-item-brand')}>{hit.brand_name}</i>
    <span className={cx('list-item-price')}>{getDisplayPrice(hit.price)}</span>
  </div>
);

ProductHit.propTypes = {
  transformHitImageUrl: PropTypes.func,
  hit: PropTypes.shape({
    brand_name: PropTypes.string.isRequired,
    objectID: PropTypes.string.isRequired,
    price: PropTypes.number,
    primary_image_url: PropTypes.string.isRequired,
    product_path: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
  })
};

export default ProductHit;
