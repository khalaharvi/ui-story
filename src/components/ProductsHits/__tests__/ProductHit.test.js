import React from 'react';
import { shallow } from 'enzyme';

import ProductHit from '../ProductHit';

describe('<ProductHit />', () => {
  it('renders correctly', () => {
    const hit = {
      brand_name: 'Brand Name',
      objectID: '12345',
      price: 1000,
      primary_image_url: 'https://placehold.it/270x270',
      product_path: '/product',
      title: 'Product Title'
    };

    const wrapper = shallow(<ProductHit hit={hit} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with transformHitImageUrl', () => {
    const hit = {
      brand_name: 'Brand Name',
      objectID: '12345',
      price: 1000,
      primary_image_url: 'https://placehold.it/270x270',
      product_path: '/product',
      title: 'Product Title'
    };

    const wrapper = shallow(
      <ProductHit
        hit={hit}
        transformHitImageUrl={url => url}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
