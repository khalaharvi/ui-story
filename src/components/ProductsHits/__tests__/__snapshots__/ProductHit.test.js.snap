// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductHit /> renders correctly 1`] = `
<div>
  <div
    className="dh-products-hits-list-item-img"
  >
    <Image
      alt="Product Title"
      src="https://placehold.it/270x270"
    />
  </div>
  <b
    className="dh-products-hits-list-item-title"
  >
    Product Title
  </b>
  <i
    className="dh-products-hits-list-item-brand"
  >
    Brand Name
  </i>
  <span
    className="dh-products-hits-list-item-price"
  >
    $1,000
  </span>
</div>
`;

exports[`<ProductHit /> renders correctly with transformHitImageUrl 1`] = `
<div>
  <div
    className="dh-products-hits-list-item-img"
  >
    <Image
      alt="Product Title"
      src="https://placehold.it/270x270"
    />
  </div>
  <b
    className="dh-products-hits-list-item-title"
  >
    Product Title
  </b>
  <i
    className="dh-products-hits-list-item-brand"
  >
    Brand Name
  </i>
  <span
    className="dh-products-hits-list-item-price"
  >
    $1,000
  </span>
</div>
`;
